pysal/esda/moran.py

```diff
@@ -188,13 +198,18 @@ def __moments(self):
         self.VI_norm = v_num / v_den - (1.0 / (n - 1)) ** 2
         self.seI_norm = self.VI_norm ** (1 / 2.)
 
-        k = (1 / ((z ** 4).sum()) * (((z ** 2).sum()) ** 2))
-        vi = (1 / (((n - 1) ** 3) * s02)) * ((n * ((n * n - 3 * n + 3)
-                                                   * s1 - n * s2 + 3 * s02))
-                                             - (k * ((n * n - n) * s1 - 2 * n *
-                                                     s2 + 6 * s02)))
-        self.VI_rand = vi
-        self.seI_rand = vi ** (1 / 2.)
+        # variance under randomization
+        xd4 = z**4
+        xd2 = z**2
+        k_num = xd4.sum() / n
+        k_den = (xd2.sum() / n)**2
+        k = k_num / k_den
+        EI = self.EI
+        A = n * ((n * n - 3 * n + 3) * s1 - n * s2 + 3 * s02)
+        B = k * ((n * n - n) * s1 - 2 * n * s2 + 6 * s02  )
```

Comment:
these are pretty hairy terms. The only things I think would be more readable here are to always refer to `n * n` as `n**2` and use `n*(n-1)` for `(n**2 - n)`, but that latter one may just be my own perception. 

Regardless, this looks okay as merged. I'm not sure the original reference on the variance under randomization though? Would that be in the Tiefelsdorf exact distribution paper?

Reply:
Good catches. Will update.

The original reference on the moments is Cliff and Ord (1981) . The link in our doctest has a worked example that reproduces those moments.